src/dg/dg_base.cpp

```diff
@@ -2619,13 +2619,15 @@ void DGBase<dim,real,MeshType>::apply_inverse_global_mass_matrix(
 
 template<int dim, typename real, typename MeshType>
 void DGBase<dim,real,MeshType>::apply_global_mass_matrix(
-        dealii::LinearAlgebra::distributed::Vector<double> &input_vector,
+        const dealii::LinearAlgebra::distributed::Vector<double> &input_vector,
         dealii::LinearAlgebra::distributed::Vector<double> &output_vector,
-        const bool use_auxiliary_eq)
+        const bool use_auxiliary_eq,
+        const bool use_M_norm)
 {
     using FR_enum = Parameters::AllParameters::Flux_Reconstruction;
     using FR_Aux_enum = Parameters::AllParameters::Flux_Reconstruction_Aux;
-    const FR_enum FR_Type = this->all_parameters->flux_reconstruction_type;
+    const FR_enum FR_cDG = FR_enum::cDG;
+    const FR_enum FR_Type = (use_M_norm) ? FR_cDG : this->all_parameters->flux_reconstruction_type;
```

Comment:
what is going on here?

Reply:
the mass matrix is assembled as a modified mass matrix, so M+K. If we want to force an application of the unmodified mass matrix M, we choose cDG (which is zero and results in K=0), rather than the C parameter which was chosen in the .prm file. I will add a comment, let me know if it should be further clarified.

Reply:
got it! thanks